server/src/main/java/io/druid/initialization/Initialization.java

```diff
@@ -78,13 +77,15 @@
 import java.util.Map;
 import java.util.ServiceLoader;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 
 /**
  */
 public class Initialization
 {
   private static final Logger log = new Logger(Initialization.class);
-  private static final Map<String, URLClassLoader> loadersMap = Maps.newHashMap();
+  private static final ConcurrentMap<File, URLClassLoader> LOADERS_MAP = new ConcurrentHashMap<>();
```

Comment:
I *think* things were more or less guarded by time or locking waaaaay up the call stack. There isn't really any explicit guarantee that `getClassLoaderForExtension` is or is not thread safe. Since this map is updated in a public static method (and pretty trivial to make threadsafe) I figured I'd make the change.